### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by Paillat-dev**
Adds the following:
- A `dockerfile`
- A `.dockerignore` file
- A `docker-compose.yaml` file
- A github action that builds the image and pushes it to `ghcr.io/itsdrike/code-jam-2024:main` on pull request

Lmk if anything is missing

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.github/
+.tests/
+.vscode/
+.editorconfig
+.gitattributes
+.gitignore
+.pre-commit-config.yaml
+LICENSE.txt
```
Comment (ItsDrike) [inline .dockerignore]: License should remain present. Even though it's technically not a useful file.
```suggestion
```

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.github/
+.tests/
+.vscode/
+.editorconfig
+.gitattributes
+.gitignore
```
Comment (ItsDrike) [inline .dockerignore:7]: We can also ignore the `.git` directory
```suggestion
.gitignore
.git
```

**Review by ItsDrike — CHANGES_REQUESTED** [review]

**Review by ItsDrike — APPROVED** [review]

**Comment by Paillat-dev** [comment]
Wait before merge

**Comment by Paillat-dev** [comment]
It's good

**Review by ItsDrike — APPROVED** [review]

**Pull request merged**